lib/rexml/element.rb

```diff
@@ -1516,6 +1508,11 @@ def write(output=$stdout, indent=-1, transitive=false, ie_hack=false)
       formatter.write( self, output )
     end
 
+    protected
+    def reset_parent_namespaces
+      @parent_namespaces = nil
+      self.each{|e| e.reset_parent_namespaces }
```

Comment:
Looks like TextNode needs to be handled correctly
```ruby
doc = REXML::Document.new('<a><a>a</a></a>')
doc.add_namespace 'foo'
#=> undefined method 'reset_parent_namespaces' for an instance of REXML::Text
```

Reply:
Oh sorry.
I will fix this case.